### PR TITLE
BUG: Mark LabelGeometryImageFilter as deprecated

### PR DESCRIPTION
**Pull request by blowekamp**
The LabelGeometryImageFilter has known computational inefficiencies and bugs such as some attributes are not computed with respect to image geometry.

This class is currently in the Review module and is planned to be moved to the Compatibility group under the Deprecated module.

## PR Checklist
- [ ] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [ ] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [ ] Added test (or behavior not changed)
- [ ] Updated API documentation (or API not changed)
- [ ] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [ ] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [ ] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKSphinxExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

**Comment by blowekamp** [comment]
The intention is to produce a warning for the current up common release so that with 5.0 the class can be moved to the deprecated module.

**Review by dzenanz — APPROVED** [review]
Looks good. You mean to be moved to `Deprecated` module in 6.0, not 5.0?

**Comment by dzenanz** [comment]
Maybe because it is off by default?

**Comment by dzenanz** [comment]
Perhaps we should turn it on by default (low effort), of finish migration of content out of it and into other corresponding modules (high effort)?

**Comment by blowekamp** [comment]
> Perhaps we should turn it on by default (low effort), of finish migration of content out of it and into other corresponding modules (high effort)?

I think turning it on by default may encourage more usage, and limit options available to the remaining classes. I Think the options for the classes in the Review module should be 1) Integrate in ITK proper after changes, 2) make a remote module 3) move to the Deprecate module.

File: Modules/Nonunit/Review/include/itkLabelGeometryImageFilter.h [hunk 1]
```diff
@@ -66,6 +66,8 @@ namespace itk
  *  by Padfield D., Miller J
  *  https://www.insight-journal.org/browse/publication/301
  *
+ * This class contains computational inefficiencies and bugs such as some attributes are not computed with respect to
+ * image geometry, consider using these supported alternatives:
```
Comment (thewtex) [inline Modules/Nonunit/Review/include/itkLabelGeometryImageFilter.h:70]: :+1:

**Comment by thewtex** [comment]
> This filter is only in ITKReview, so perhaps less notification is needed?

ITKReview is relied on in practice, but it would be helpful to have it marked as deprecated if you would like to move to ITKDeprecated in ITK 6. Please go ahead and merge.

**Pull request merged**